### PR TITLE
Update to new corefx

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs [hunk 1]
```diff
@@ -143,7 +144,7 @@ public WebSocketsTransport(WebSocketOptions options, IDuplexPipe application, De
 #if NETCOREAPP2_1
                     var receiveResult = await socket.ReceiveAsync(memory, CancellationToken.None);
 #else
-                    var isArray = memory.TryGetArray(out var arraySegment);
+                    var isArray = MemoryMarshal.TryGetArray<byte>(memory, out var arraySegment);
```
Comment (anurse) [inline src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs]: Is the `<byte>` not inferrable?

Comment (pakrym) [reply src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs]: The first parameter type is `ReadOnlyMemory`, it's either inference or implicit conversion, unfortunately.

**Comment by pakrym** [comment]
I hope so, it contains the fix for the issue OSX we reported. We'll see if it works :)

File: src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs [hunk 2]
```diff
@@ -367,7 +367,7 @@ private static Uri CreateConnectUrl(Uri url, string connectionId)
         private async Task StartTransport(Uri connectUrl, TransportType transportType)
         {
 
-            var options = new PipeOptions(readerScheduler: PipeScheduler.ThreadPool);
+            var options = new PipeOptions(writerScheduler:PipeScheduler.Inline, readerScheduler: PipeScheduler.ThreadPool, useSynchronizationContext: false);
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs]: nit: space after `:`

File: src/Common/PipeWriterStream.cs [hunk 3]
```diff
@@ -63,11 +63,11 @@ public override Task WriteAsync(byte[] buffer, int offset, int count, Cancellati
         }
 
 #if NETCOREAPP2_1
-        public override Task WriteAsync(ReadOnlyMemory<byte> source, CancellationToken cancellationToken = default)
+        public override ValueTask WriteAsync(ReadOnlyMemory<byte> source, CancellationToken cancellationToken = default)
         {
             _pipeWriter.Write(source.Span);
             _length += source.Length;
-            return Task.CompletedTask;
+            return new ValueTask(Task.CompletedTask);
```
Comment (davidfowl) [inline src/Common/PipeWriterStream.cs]: return default;

File: src/Common/WebSocketExtensions.cs [hunk 4]
```diff
@@ -20,7 +20,7 @@ public static Task SendAsync(this WebSocket webSocket, ReadOnlySequence<byte> bu
 #if NETCOREAPP2_1
             if (buffer.IsSingleSegment)
             {
-                return webSocket.SendAsync(buffer.First, webSocketMessageType, endOfMessage: true, cancellationToken);
+                return webSocket.SendAsync(buffer.First, webSocketMessageType, endOfMessage: true, cancellationToken).AsTask();
```
Comment (davidfowl) [inline src/Common/WebSocketExtensions.cs]: Make this method return ValueTask.

**Comment by davidfowl** [comment]
Super small set of changes.

**Pull request merged**